pandas/io/parsers/readers.py

```diff
@@ -1515,6 +1515,11 @@ def _clean_options(
             if engine == "c" and sep == r"\s+":
                 result["delim_whitespace"] = True
                 del result["delimiter"]
+            elif engine == "pyarrow" and (sep in ("\\s+", "  ", "||")):
+                fallback_reason = (
+                    "The 'pyarrow' engine does not support separators with more than "
+                    "one character. Please use the C parser instead."
```

Comment:
From the documentation doesn't seem like the `c` engine supports more than one character except for `\s+`. I'd better say `Please consider using a different engine`.